### PR TITLE
Unpack Kubernetes config

### PR DESCRIPTION
**Pull request by pidster**
Fixes: https://github.com/microservices-demo/microservices-demo/issues/310

### CODE REVIEW

File: deploy/kubernetes/manifests/v1/ksNamespace.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kube-system
```
Comment (errordeveloper) [inline deploy/kubernetes/manifests/v1/ksNamespace.yaml]: I kept wondering why do we have to define this?

**Comment by errordeveloper** [comment]
More comments to follow, but overall I am not sure about having `v1` and `v1beta1` is so appropriate. It seems as if versioned directories refer to application versions, and they aren't...

File: deploy/kubernetes/manifests/complete-demo.yaml [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: user
+  labels:
+    name: user
+spec:
+  ports:
+    # the port that this service should serve on
```
Comment (errordeveloper) [inline deploy/kubernetes/manifests/complete-demo.yaml]: These comments seem quite redundant, especially because we aren't doing anything worse explaining here, and it's not like we comment on other fields (if we were to attempt to explain Kubernetes).

File: deploy/kubernetes/manifests/weavescope.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+apiVersion: v1
```
Comment (errordeveloper) [inline deploy/kubernetes/manifests/weavescope.yaml]: Why do we need this file here? The docs reference launcher URL, which is what we should do.

**Review by errordeveloper — COMMENTED** [review]
Overall this is much better then how we had all of these in one directory. I am not sure about having `v1` and `v1beta1` is so appropriate. It seems as if versioned directories refer to application versions, and they aren't... Could you move those into one subdirectory and simply call it `app`, with `compete-demo.yaml` renamed to `app-combined.yaml` to match?

File: deploy/kubernetes/manifests/v1/ksNamespace.yaml (hunk 1 above)
Comment (errordeveloper) [reply deploy/kubernetes/manifests/v1/ksNamespace.yaml]: I think it's an artefact of something that we didn't have to do the first place, also the way namespaces are created has changed recently also. It's okay keep it, if we have scripts that refer to it, and remove when we move to `kubeadm`...

File: deploy/kubernetes/manifests/catalogue-db-dep.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+apiVersion: extensions/v1beta1
```
Comment (squaremo) [inline deploy/kubernetes/manifests/catalogue-db-dep.yaml:2]: In general it's useful to put `---` at the top of yamels, so they can be 'cated together.

**Review by errordeveloper — APPROVED** [review]
LGTM, let's discuss whether we need `weavescope.yaml` separately.

File: deploy/kubernetes/weavescope.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
+apiVersion: v1
```
Comment (squaremo) [inline deploy/kubernetes/weavescope.yaml:1]: Is this supposed to be here?

I don't think it'll work with fluxy, which expects either Deployment or ReplicationController resources. But it might not break it.

File: deploy/mesos-marathon/marathon.json [hunk 6]
```diff
@@ -5,18 +5,18 @@
       "id": "/weave-demo/database",
```
Comment (squaremo) [inline deploy/mesos-marathon/marathon.json:5]: Does this change definitely belong in this PR? It seems unrelated.

File: deploy/kubernetes/complete-demo.yaml [hunk 7]
```diff
@@ -0,0 +1,437 @@
+apiVersion: extensions/v1beta1
```
Comment (squaremo) [inline deploy/kubernetes/complete-demo.yaml:1]: I don't think you need to keep the all-in-one file around.

**Comment by errordeveloper** [comment]
The combined manifest is very handy for demos, you just pass Github URL to
kubectl and that's all.

File: deploy/kubernetes/complete-demo.yaml (hunk 7 above)
Comment (pidster) [reply deploy/kubernetes/complete-demo.yaml:1]: The all-in-one is separate use-case, as @errordeveloper says.

File: deploy/mesos-marathon/marathon.json (hunk 6 above)
Comment (pidster) [reply deploy/mesos-marathon/marathon.json:5]: Rebase pulled it in.

File: deploy/kubernetes/weavescope.yaml (hunk 5 above)
Comment (pidster) [reply deploy/kubernetes/weavescope.yaml:1]: Ack. Not applicable to flux.

File: deploy/kubernetes/manifests/catalogue-db-dep.yaml (hunk 4 above)
Comment (pidster) [reply deploy/kubernetes/manifests/catalogue-db-dep.yaml:2]: Noted, will add this.

**Comment by mongrelion** [comment]
Hey, guys, I noticed that this build is failing because of a problem in the `openapi/Dockerfile` file, which has been fixed in #314. Can you update this branch with master? that should fix this problem and your build should be green, then.
